starlette/http_exception.py

```diff
@@ -0,0 +1,20 @@
+import http
+import typing
+
+
+class HTTPException(Exception):
+    def __init__(
+        self,
+        status_code: int,
+        detail: typing.Optional[str] = None,
+        headers: typing.Optional[dict] = None,
+    ) -> None:
+        if detail is None:
+            detail = http.HTTPStatus(status_code).phrase
+        self.status_code = status_code
+        self.detail = detail
+        self.headers = headers
+
+    def __repr__(self) -> str:
+        class_name = self.__class__.__name__
+        return f"{class_name}(status_code={self.status_code!r}, detail={self.detail!r})"
```

Comment:
Why do we need to move this?

Reply:
Circular import. `exceptions -> requests`; `requests -> formparsers`; `formparsers -> exceptions`

Reply:
Gotcha. I was always confused as to why there was a middleware in the exceptions module. Could we move the middleware instead?

Reply:
hmm... I didn't think about it. But it makes sense. 👍

Reply:
might be nice to "deprecate" it from this module as well even if we re-export it in the short term

Reply:
Ok. That's probably the reason I didn't thought about it. With the relocation of `HTTPException`, the import remains here, so there's no need to deprecate it. I'd have to introduce the `__dir__` and `__getattr__` on this file as well.

Reply:
I think that's worth it right? I really do feel like the middleware should be in `starlette.middleware.exceptions`

Reply:
Not sure 😅

Reply:
Fair enough, it's your call now that you've tried it both ways!